src/ansible/roles/keycloak/tasks/main.yml

```diff
@@ -0,0 +1,148 @@
+- name: print keycloak_url
+  debug:
+    msg: "keycloak_url={{ keycloak_url }}"
+
+- name: Download software zipfile
+  get_url:
+    url: "{{ keycloak_url }}"
+    dest: /tmp
+    
+- name: Unzip software
+  unarchive:
+    remote_src: yes
+    src: "/tmp/{{ zip_file }}"
+    dest: /tmp
+
+- name: Check if unzipped directory exists
+  stat:
+    path: "{{ unzip_dir }}"
+  register: unzip_dir_chk
+
+- name: debug unzip dir
+  debug:
+    var: unzip_dir_chk.stat
+
+- name: Check if /opt/keycloak dir exists
+  stat:
+    path: /opt/keycloak
+  register: opt_keycloak_chk
+
+- name: debug opt dir
+  debug:
+    var: opt_keycloak_chk.stat
+
+- name: Move software to /opt
+  command: mv {{ unzip_dir }} /opt/keycloak 
+  when: unzip_dir_chk.stat.exists and not opt_keycloak_chk.stat.exists
```

Comment:
You can extract it directly to /opt/keycloak.

Also IMHO you don't need to care about impotence here. You can just override the folder if it exist - which it never does in containers.

Reply:
Extracts directly to dir now.